aas-gui/Frontend/aas-web-gui/src/components/SubmodelPlugins/CarbonFootprint.vue

```diff
@@ -0,0 +1,180 @@
+    components: {
+        RequestHandling, // Mixin to handle the requests to the AAS
+    },
+    mixins: [RequestHandling, SubmodelElementHandling],
+    props: ['submodelElementData'],
+    setup() {
+        const theme = useTheme();
+        const navigationStore = useNavigationStore();
+        const aasStore = useAASStore();
+
+        return {
+            theme, // Theme Object
+            navigationStore, // NavigationStore Object
+            aasStore, // AASStore Object
+        };
+    },
+    data() {
+        return {
+            productCarbonFootprintData: {} as Record<string, any>,
+            transportCarbonFootprintData: {} as Record<string, any>,
+            productIsoValue: '' as string,
+            transportENValue: '' as string,
+        };
+    },
+    mounted() {
+        // console.log('Component Mounted. SubmodelElementData:', this.submodelElementData);
+        this.initializeProductCarbonFootprint(); // initialize Product Carbon Footprint Plugin
+        this.initializeTransportCarbonFootprint(); // initialize Transport Carbon Footprint Plugin
+    },
+    computed: {
+        SelectedNode() {
+            return this.aasStore.getSelectedNode;
+        },
+        pcfPercentage() {
+            return this.calculatePercentage(this.productIsoValue, this.transportENValue);
+        },
+        tcfPercentage() {
+            return this.calculatePercentage(this.transportENValue, this.productIsoValue);
+        },
+        pcfValueColor() {
+            return this.getThemeColor('pcf');
+        },
+        tcfValueColor() {
+            return this.getThemeColor('tcf');
+        },
+        fingersColor() {
+            return this.getThemeColor('fingers');
+        }
+    },
+    methods: {
+        calculatePercentage(value1: string, value2: string): number {
+            const v1 = parseFloat(value1) || 0;
+            const v2 = parseFloat(value2) || 0;
+            const total = v1 + v2;
+            return total ? (v1 / total) * 100 : 0;
+        },
+        getThemeColor(colorKey: string): string {
+            const isDark = this.$vuetify.theme.global.current.dark;
+            const themeColors = isDark ? this.$vuetify.theme.themes.dark.colors : this.$vuetify.theme.themes.light.colors;
+            return themeColors[colorKey];
+        },
+        // Utility function to initialize Carbon Footprint
+        initializeCarbonFootprint(type: string, dataKey: 'productCarbonFootprintData' | 'transportCarbonFootprintData', valueKey: 'productIsoValue' | 'transportENValue') {
+            this[dataKey] = {};
+            this[valueKey] = "";
+
+            if (Object.keys(this.submodelElementData).length === 0) {
+                return;
+            }
+
+            try {
+                const carbonFootprintData = { ...this.submodelElementData };
+                const entries = carbonFootprintData.submodelElements.filter((elem: any) => elem.idShort && elem.idShort.includes(type));
```

Comment:
You are checking if the type matches here which is the constant `RODUCT_CARBON_FOOTPRINT` (`TRANSPORT_CARBON_FOOTPRINT`).
In reality, the PCF (TCF) SubmodelElementCollection could be called something else. You should rather select it by its SemanticId.